vyper/semantics/analysis/local.py

```diff
@@ -801,3 +736,82 @@ def visit_IfExp(self, node: vy_ast.IfExp, typ: VyperType) -> None:
         self.visit(node.body, typ)
         validate_expected_type(node.orelse, typ)
         self.visit(node.orelse, typ)
+
+
+def _analyse_range_call(node: vy_ast.Call) -> list[VyperType]:
+    """
+    Check that the arguments to a range() call are valid.
+    :param node: call to range()
+    :return: None
+    """
+    validate_call_args(node, (1, 2), kwargs=["bound"])
+    kwargs = {s.arg: s.value for s in node.keywords or []}
+
+    start, end = (vy_ast.Int(value=0), node.args[0]) if len(node.args) == 1 else node.args
+
+    if "bound" in kwargs:
+        return _analyse_range_bound(start, end, kwargs["bound"], node)
+
+    return _analyse_range_constant(start, end, node)
```

Comment:
i think the code paths for `analyse_range_bound()` and `analyse_range_constant()` can somehow be merged together now